[ProductDownloadsView] Restricting "Featured Tutorials" section to just tutorials

ashleemboyer opened the pull request:
## 🔗 Relevant links

<!--
Include links to the branch preview, Asana task, and Figma designs wherever possible to make reviewing your PR easier.
When including the preview link, make sure to remove any '.' characters from the branch name:
  - Example: ks.my-branch -> ksmy-branch
-->

- [Vault Preview link](https://dev-portal-git-ambrestrict-featured-tutorials-section-hashicorp.vercel.app/vault/downloads) 🔎
- [Waypoint Preview link](https://dev-portal-git-ambrestrict-featured-tutorials-section-hashicorp.vercel.app/waypoint/downloads) 🔎
- [Asana task](https://app.asana.com/0/1202702246158644/1202780278912298/f) 🎟️

## 🗒️ What

<!--
Briefly list out the changes proposed in this PR.
-->

- Updates the Feature Tutorials section of the downloads views to only show tutorials
- Updates the Vault and Waypoint sections to have 6 tutorials

## 🤷 Why

<!--
Describe why the changes proposed are needed. Some examples: new feature requested, refactor to make things easier later, styling tweaks requested by design, etc.
-->

- We do not want to mix and match types of cards, especially agter bookmarking and progress are implemented.

## 🧪 Testing

<!--
Create a checklist for going through how to test your proposed changes. If there is anything to configure before interacting with the project in a browser, such as toggling feature flags, changing machine settings, or simulating behavior in browser dev tools, list those steps first.

- [ ] Step 1
- [ ] Step 2
- [ ] Step 3
- [ ] ...
-->

- [ ] Go to [/vault/downloads](https://dev-portal-git-ambrestrict-featured-tutorials-section-hashicorp.vercel.app/vault/downloads)
- [ ] Make sure the Featured Tutorials content is OK
- [ ] Go to [/waypoint/downloads](https://dev-portal-git-ambrestrict-featured-tutorials-section-hashicorp.vercel.app/waypoint/downloads)
- [ ] Make sure the Featured Tutorials content is OK

BRKalow reviewed (approved):
👍 👍

tunzor reviewed (approved):
Waypoint content looks good!

tunzor commented on src/content/waypoint/install-landing.json:
```diff
@@ -1,23 +1,11 @@
-		}
+	"featuredTutorialsSlugs": [
+		"waypoint/get-started-intro",
+		"waypoint/get-started-install",
```
This actually refers to the kubernetes guide at `waypoint/tutorials/get-started-kubernetes/get-started-install` but it's picking up the docker one (which is the first linked guide in the list).

Will something like this work?
```suggestion
		"waypoint/get-started-kubernetes/get-started-install",
```

ashleemboyer replied:
@kendallstrautman @BrandonRomano is it possible to specify which collection we'd like to link to here if it's not a tutorial's default collection? These tutorial slugs are passed to `featuredTutorialsSlugs`, which calls `getTutorials`. I'm not familiar enough with that endpoint to know if it will accept a tutorial slug that includes information about the collection.

tunzor reviewed (commented):
Sorry, noticed the link to the wrong tutorial after I clicked approve!
See the comment above for the suggestion.

ashleemboyer commented:
@BRKalow @tunzor I've updated the implementation to allow two formats of slugs in this section:

- `:productSlug/:tutorialSlug`
- `:productSlug/:collectionSlug/:tutorialSlug`

I've also updated the docs so this is in the instructions for authoring this page. What do y'all think?

tunzor commented:
> @BRKalow @tunzor I've updated the implementation to allow two formats of slugs in this section:
> 
> * `:productSlug/:tutorialSlug`
> * `:productSlug/:collectionSlug/:tutorialSlug`
> 
> I've also updated the docs so this is in the instructions for authoring this page. What do y'all think?

Perfect!

BRKalow reviewed (approved):
🎉
